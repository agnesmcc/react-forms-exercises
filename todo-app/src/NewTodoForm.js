import React from "react";

const NewTodoForm = ({addTodo}) => {
    const INITIAL_STATE = {
        task: ""
    }
    const [formData, setFormData] = React.useState(INITIAL_STATE)

    const handleSubmit = (e) => {
        e.preventDefault();
        console.log(formData)
        addTodo({...formData})
        setFormData(INITIAL_STATE)
        document.getElementById('task').focus();
    }   

    const handleChange = (e) => {
        const {name, value} = e.target
        setFormData(formData => ({
            ...formData,
            [name]: value
        }))
    }   

    return (
        <div>   
            <form onSubmit={handleSubmit}>
                <div>
                    <label htmlFor="task">Task</label>
                    <input id="task" name="task" type="text" 
                        value={formData.task} 
                        onChange={handleChange}/>
                </div>  
                <button>Add a new todo</button>
            </form>
        </div>
    )
}   

export default NewTodoForm;
