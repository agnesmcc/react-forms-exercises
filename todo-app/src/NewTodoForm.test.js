import { render } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

it('renders without crashing', () => {
  const { getByText } = render(<NewTodoForm />);
  const input = getByText(/Task/);
  expect(input).toBeInTheDocument();
})

it('matches the snapshot', () => {
  const { asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
}) 
