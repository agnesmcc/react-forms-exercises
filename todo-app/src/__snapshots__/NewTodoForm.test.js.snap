// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <div>
        <label
          for="task"
        >
          Task
        </label>
        <input
          id="task"
          name="task"
          type="text"
          value=""
        />
      </div>
      <button>
        Add a new todo
      </button>
    </form>
  </div>
</DocumentFragment>
`;
